_posts/2016-10-31-Kickstarters-Project.md

```diff
@@ -0,0 +1,205 @@
+For example people leaving the TV on for too long inside of the meeting rooms.
+Also employees are able to see if the room is occupied or not, because right now booking a room with outlook isn't working so well as a lot of you might already have experienced.
+
+
+### Used technologies - sofware
+
+* back-end              
+  * Maven
+  * Spring
+  * SpringBoot
+  * MongoDB
+  * Mockito
+
+* front-end
+  * Angular 2
+  * Angular Material 2
+
+* extra
+  * Waffle
+  * Cloudfoundry
+  * GitHub
+
+
+### process of the project
+
+The first week went pretty well, they devided themselves up into 2 groups.
+One group for the front-end and the other one for the back-end.
+At first they started to create different user stories for their SCRUM board.
+Using the newly acquired scrum techniques during the first month.
+The kickstarters had to work in 4 short sprints of one week, during the first sprint they also decided to change their real life SCRUM board into an online version named Waffle.
+This software would track the pull requests and merges automatically from Github and change the board accordingly.
+Continuous Integration was pretty important during the course of the project.
+This way whenever they made changes to the code and made a pull request to Github that failed to build, they had to fix their code before they could continue.
+Once the build on codeship succeeded and the pull request was merged. The main development branch and master branch would have their changes (if any) deployed to their Cloudfoundry instance.
+
+The process for the back-end was pretty simple.
+They started out with around 5 people, so some of them started to pair program while others started to program on their own.
+But, with paying attention to the SCRUM principles and how to write clean code.
+So starting with the basic implementations of the sensor, room and notifications classes and writing the JUnit and Mockito tests was their first job.
+While the back-end was pretty straight forward and relatively easy to begin with, the front-end team were confronted with some problems.
+The team consisted of only 2 people who had to tackle a lot of problems with the use of Angular 2 which was still in Beta at the time.
+Also the combination with the other frameworks wasn't quite that easy to work with.
+
+Every morning the team did a stand up meeting where they would discuss their changes in eachothers code and what they were going to do next.
+During these 4 weeks they tried to work in these short sprints but after a week or two it became clear this wouldn't be an easy task.
+A few team members already had left the group because they were assigned to projects, which messed up their sprints completely.
+
+During the first 2 weeks the back-end team started with using Spring Data JPA, but soon figured out Spring-boot-starter-data-MongoDB was the better alternative because of the large amount of data being pumped into the DB.
```

Comment:
Replace`spring-boot-starter-data-mongodb` with `Spring Data MongoDB` to be inline with `Spring Data JPA`.